src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/ArrayMarshaller.cs

```diff
@@ -22,6 +22,12 @@ namespace System.Runtime.InteropServices.Marshalling
     public static unsafe class ArrayMarshaller<T, TUnmanagedElement>
         where TUnmanagedElement : unmanaged
     {
+        /// <summary>
+        /// Allocate memory for the unmanaged representation of the collection.
```

Comment:
Should we go with `array` instead of `collection`, since this marshaller is specifically for arrays?